### PR TITLE
[DCJ-220][DCJ-184][DCJ-182] Refactor react-hyperscript-helper components to JSX

### PR DESCRIPTION
**Pull request by PintoGideon**
### Addresses

https://broadworkbench.atlassian.net/browse/DCJ-220
https://broadworkbench.atlassian.net/browse/DCJ-184
https://broadworkbench.atlassian.net/browse/DCJ-182

### Summary

1. Remove the use of react-hyperscript-helpers in the Dataset Catalog page
2. Update the file extensions from js to jsx

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/DatasetCatalog.jsx [hunk 1]
```diff
@@ -0,0 +1,777 @@
...
+          <div className={currentUser.isAdmin ? style['table-scroll-admin'] : style['table-scroll']}>
+            <table className="table">
+              <thead>
+                <tr>
+                  <th />
+                  {currentUser.isAdmin || currentUser.isChairPerson && (
+                    <th className={style['cell-size']} style={{ minWidth: '14rem' }}>Actions</th>
+                  )}
+                  <th className={style['cell-size']}>{getSortDisplay({ field: 'datasetIdentifier', label: 'Dataset ID' })}</th>
+                  <th className={style['cell-size']}>{getSortDisplay({ field: 'Dataset Name' })}</th>
```
Comment (rushtong) [inline src/pages/DatasetCatalog.jsx]: There is an off-by-one error somewhere here for the admin case. Note in the screenshot how the Dataset ID column has actions and the Dataset Name column has the ID, etc.

## Admin Case
Screenshot 2024-04-29 at 8 24 02 AM

## Non-Admin case
Screenshot 2024-04-29 at 8 27 27 AM

**Review by rushtong — CHANGES_REQUESTED** [review]
Looks good aside from one bug - see comment inline, thank you!

File: src/pages/DatasetCatalog.jsx (hunk 1 above)
Comment (PintoGideon) [reply src/pages/DatasetCatalog.jsx]: @rushtong  Thanks for catching this.

**Review by rushtong — APPROVED** [review]
Thank you! 👍🏽

**Review by fboulnois — APPROVED** [review]
looks good 👍

**Pull request merged**